Source/Snowman.cpp

```diff
@@ -273,3 +290,135 @@ void Snowman::randomTranslationSnowman(GLFWwindow* window, const bool& shift, bo
 	}
 	canRandomPlacement = !randomPlacement;
 }
+
+/**
+* Check collision of the snowman to another collider object using its vec3
+* @return true if the object is colliding & false otherwise.
+*/
+bool Snowman::CheckCollision(std::vector<CollidableModel>& colliders) {
+	bool collisionX = false;
+	bool collisionZ = false;
+
+	for (auto& collider : colliders) {
+		glm::vec3 positionCollider = collider.getColliderPosition();
+		glm::vec3 scaleCollider = collider.getColliderScale();
+
+		collisionX = (origin.x + this->getDimensions().x >= positionCollider.x && // snowman going from right to left is colliding
+					  positionCollider.x + collider.getCollidableDimensions().x >= origin.x) ||
+					 (origin.x - this->getDimensions().x <= positionCollider.x && // snowman going from left to right is colliding
+					  positionCollider.x - collider.getCollidableDimensions().x <= origin.x);
+
+		collisionZ = (origin.z + this->getDimensions().z >= positionCollider.z && // snowman going forward is colliding
+					  positionCollider.z + collider.getCollidableDimensions().z >= origin.z) ||
+					 (origin.z - this->getDimensions().z <= positionCollider.z && // snowman going backwards is colliding
+					  positionCollider.z - collider.getCollidableDimensions().z <= origin.z);
+
+		if (collisionX && collisionZ) {
+			colliders = { collider }; // only keep the colliding object
+			break;
+		}
+	}
+
+	isFreeLeft      = true; // for some reason it breaks if i don't add these lines
+	isFreeBackwards = true;
+	isFreeRight     = true;
+	isFreeForwards  = true;
+
+	//std::cout << collisionX << " " << collisionZ << std::endl;
+	return collisionX && collisionZ;
+}
+
+/**
+* If the snowman is colliding, check if it is colliding on the x axis
+* Disabled the key associated that is going in the direction of the collision
+*/
+void Snowman::CheckCollisionX(std::vector<CollidableModel> colliders, bool isColliding) {
+
+	for (auto& collider : colliders) {
+		glm::vec3 positionCollider = collider.getColliderPosition();
+
+		// keyA
+		if (origin.x <= positionCollider.x && // check if the snowman is colliding with the right side (between origin and half its width)
+			origin.x + this->getDimensions().x / 2 >= positionCollider.x - collider.getCollidableDimensions().x / 2 && isColliding) {
+			isFreeLeft  = false;
+			isFreeRight = true;
```

Comment:
Something is fishy about the CheckCollisionX method. Here, this condition checks if the snowman is colliding with the right side--and the logic seems to me to be doing that--, only that the isFreeRight boolean is set to be TRUE instead of FALSE. Compare the CheckCollisionY method, which if it check if colliding forwards, sets isFreeForwards = FALSE

Reply:
Might've messed up the naming convention, I'll double check